test/tests/functional/pbs_mom_dynamic_resource_ext.py

```diff
@@ -0,0 +1,212 @@
+# coding: utf-8
+
+# Copyright (C) 1994-2018 Altair Engineering, Inc.
+# For more information, contact Altair at www.altair.com.
+#
+# This file is part of the PBS Professional ("PBS Pro") software.
+#
+# Open Source License Information:
+#
+# PBS Pro is free software. You can redistribute it and/or modify it under the
+# terms of the GNU Affero General Public License as published by the Free
+# Software Foundation, either version 3 of the License, or (at your option) any
+# later version.
+#
+# PBS Pro is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.
+# See the GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+# Commercial License Information:
+#
+# For a copy of the commercial license terms and conditions,
+# go to: (http://www.pbspro.com/UserArea/agreement.html)
+# or contact the Altair Legal Department.
+#
+# Altair’s dual-license business model allows companies, individuals, and
+# organizations to create proprietary derivative works of PBS Pro and
+# distribute them - whether embedded or bundled with other software -
+# under a commercial license agreement.
+#
+# Use of Altair’s trademarks, including but not limited to "PBS™",
+# "PBS Professional®", and "PBS Pro™" and Altair’s logos is subject to Altair's
+# trademark licensing policies.
+
+from tests.functional import *
+
+
+class TestMomDynRes(TestFunctional):
+    def create_mom_dyn_resc(self, resc_name,
+                            resc_type, resc_flag,
+                            script_body):
+        """
+        helper function for create mom_dyn_res.
+        """
+        attr = {"type": resc_type, "flag": resc_flag}
+        self.server.manager(MGR_CMD_CREATE, RSC, attr,
+                            id=resc_name, expect=True)
+
+        filename = "foobar"
+        tmpfilepath = os.path.join(os.sep, "tmp", filename)
+        with open(tmpfilepath, "w") as fd:
+            fd.write(script_body)
+        self.du.chmod(path=tmpfilepath, mode=0755)
```

Comment:
Now that I'm looking at this again, dshutils now provides the ability to create temp files.  Use self.du.create_temp_file().  You can even provide it the body of your script, so you don't have to do the write.  It'll do everything for you.  It returns the path to the file.

Reply:
@bhroam , i think you are looking into first commit. in second commit i am using "self.du.mkstemp" according to the Anamika's comment. please let me know if "self.du.mkstemp" is ok or i need to change to "self.du.create_temp_file()".

Reply:
Oh, I must have been looking at the older commit.  I think self.du.create_temp_file() is a better solution because it will do everything you need.  You won't need to open the file and write to the file.  You just pass create_temp_file() the body of the file you want, and it returns the path to the newly created temp file.  It even closes the file descriptor that mkstemp() returns.